Source/WebCore/loader/archive/Archive.cpp

```diff
@@ -50,4 +50,27 @@ void Archive::clearAllSubframeArchives(HashSet<Archive*>& clearedArchives)
     m_subframeArchives.clear();
 }
 
+bool Archive::saveResourcesToDisk(const String& directory)
+{
+    ASSERT(!RunLoop::isMain());
+
+    if (!m_mainResource)
+        return false;
+
+    if (!m_mainResource->saveToDisk(directory))
+        return false;
+
+    for (auto subresource : m_subresources) {
+        if (!subresource->saveToDisk(directory))
+            return false;
+    }
+
+    for (auto subframeArchive : m_subframeArchives) {
+        if (!subframeArchive->saveResourcesToDisk(directory))
+            return false;
+    }
```

Comment:
When we return `false`, I'm curious how we clean up partially written output.

I didn't see any code to clean up partially written output, so I think we need to do something here--unless it's always written to a temp directory that can be cleaned up by the system.